sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -236,6 +236,11 @@ def receive(self, batch=False, max_batch_size=300, max_wait_time=None):
                     # If optional dependency is not installed, do not retry.
                     if isinstance(exception, ImportError):
                         raise exception
+
+                    # If authentication exception, do not retry.
+                    if isinstance(exception, self._amqp_transport.AUTHENTICATION_EXCEPTION):
```

Comment:
do we not want to add the same to consumer_async? to not retry auth -- or will it not retry anyway for all auth exceptions (b/c while loop)?

Reply:
nvm I forgot tasks exist :)

Reply:
for async, this bit of code exists in the transports, so I added it to uamqp/pyamqp transport files